packages/amplify_authenticator/lib/src/mixins/authenticator_username_field.dart

```diff
@@ -129,23 +131,33 @@ mixin AuthenticatorUsernameField<FieldType,
               return ToggleButtons(
                 borderWidth: buttonBorderWidth,
                 constraints: buttonConstraints,
-                isSelected: [useEmail.value, !useEmail.value],
+                isSelected: [
+                  state.usernameSelection == UsernameSelection.email,
+                  state.usernameSelection == UsernameSelection.phoneNumber,
+                ],
                 onPressed: (int index) {
-                  bool useEmail = index == 0;
-                  setState(() {
-                    this.useEmail.value = useEmail;
-                  });
+                  final newUsernameSelection = index == 0
+                      ? UsernameSelection.email
+                      : UsernameSelection.phoneNumber;
+                  final oldUsernameSelection = state.usernameSelection;
+                  if (oldUsernameSelection == newUsernameSelection) {
+                    return;
+                  }
+                  state.usernameSelection = newUsernameSelection;
                   // Reset current username value to align with the current switch state.
-                  String newUsername = useEmail
+                  String newUsername = newUsernameSelection ==
+                          UsernameSelection.email
                       ? state.getAttribute(CognitoUserAttributeKey.email) ?? ''
                       : state.getAttribute(
                               CognitoUserAttributeKey.phoneNumber) ??
                           '';
                   state.username = newUsername;
```

Comment:
Since `.username` and `.usernameSelection` both trigger rebuilds, I think setting the username first might make more sense?